src/Heatmap/HeatMapInterpolationPoints.swift

```diff
@@ -0,0 +1,399 @@
+/* Copyright (c) 2020 Google Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import Foundation
+import GooglePlaces
+import GoogleMaps
+import GoogleMapsUtils
+
+/// This class will create artificial points in surrounding locations with appropriate intensities interpolated by neighboring intensity values.
+class HeatMapInterpolationPoints {
+    
+    /// The input data set; each entry contains the coordinates and the 
+    private var data = [[Double]]()
+    
+    /// The heat map layer
+    private let heatMapLayer: GMUHeatmapTileLayer = GMUHeatmapTileLayer()
+    
+    /// The heat map colors and gradient
+    private var gradientColors = [UIColor.green, UIColor.red]
+    private var gradientStartPoints = [0.005, 0.7] as [NSNumber]
+    
+    /// The user should be able to access thisdata as well as the size of the data
+    public var heatMapPoints = [GMUWeightedLatLng]()
```

Comment:
I disagree that this should be accessible. Consider returning this as an output from a method instead (e.g. `interpolate()`?).

Reply:
Sounds good! The generateHeatMaps function will set the heat map to the mapView and will also return the set of map points as a by-product.